### PR TITLE
Updater 

### PR DESCRIPTION
**Pull request by AndrewSazonov**
Work related to:
* https://github.com/easyScience/easyDiffractionApp/issues/28
* https://github.com/easyScience/easyDiffractionApp/issues/81
* https://github.com/easyScience/easyDiffractionApp/issues/58

Extra:
* `easyAppGui` and `easyAppLogic` are now merged into `easyApp`
* Redesigned application preferences dialog

### CODE REVIEW

File: easyApp/Logic/Maintenance.py [hunk 1]
```diff
@@ -189,27 +200,33 @@ def _getReleaseNotes(self):
         release_notes = re.sub(r'^# ', "### ", release_notes)
         return release_notes
 
+    def _getWebDate(self):
+        web_changelog = self._getWebChangelog()
+        pattern = r'^# Version .*? \(([A-Za-z0-9\s]*)\)'
+        matches = re.findall(pattern, web_changelog)
+        web_date = matches[0]
+        return web_date
+
```
Comment (rozyczko) [inline easyApp/Logic/Maintenance.py:209]: `QApplication.quit()` in `installUpdate` in line 70 will need user confirmation if state is changed.

**Review by rozyczko — APPROVED** [review]
Minor issues remain to be fixed but the main functionality is working fine.
One more general observation: when the maintenance tool is running, attempting to use the `Quit` button still presents me with the UAC dialog "Would you like the application to make changes to your computer?" despite saying no. There is no installation going on, just the annoying dialog which should not be there.

File: easyApp/Logic/Maintenance.py [hunk 2]
```diff
@@ -30,13 +31,9 @@ def __init__(self, parent=None):
...
         self._web_version = ""
-
+        self._web_date = ""
         self._release_notes = ""
-
-        self._process = QProcess()
-        self._process.setWorkingDirectory(QApplication.applicationDirPath())
-        #self._process.setWorkingDirectory("/Applications/easyDiffraction/MaintenanceTool.app/Contents/MacOS")
-        self._process.setProgram(Updater.exeRelativePath())
+        self._process = self._createUpdaterProcess()
```
Comment (rozyczko) [inline easyApp/Logic/Maintenance.py:36]: Should the process be initialized here? This means its lifetime is determined by the lifetime of the whole application and its instantiation here leads to longer startup time.
However, keeping track of running instances (making sure you only run it once) and subsequent checks should be noticeably faster, so maybe there is a reason for reusing the instance?

**Comment by rozyczko** [comment]
When running the update, the component display of items to be updated is less than optimal:

app

File: easyApp/Gui/Components/PreferencesDialog.qml [hunk 3]
```diff
@@ -111,23 +129,30 @@ EaElements.Dialog {
...
+                id: checkOnAppStartRow
+                spacing: EaStyle.Sizes.fontPixelSize
 
-            EaElements.CheckBox {
-                id: updatesCheckBox
-                checked: EaGlobals.Variables.checkUpdateOnAppStart
-                onCheckedChanged: EaGlobals.Variables.checkUpdateOnAppStart = checked
+                EaElements.Label {
+                    anchors.verticalCenter: parent.verticalCenter
+                    text: qsTr("Check on app start") + ":"
```
[the remark this answers is not in the document]
Comment (AndrewSazonov) [reply easyApp/Gui/Components/PreferencesDialog.qml:142]: Fixed

File: easyApp/Logic/Maintenance.py [hunk 4]
```diff
@@ -147,8 +156,10 @@ def _onFinished(self, exit_code: int, exit_status: QProcess.ExitStatus):
         # New version is found
         print(f"* Updater found component(s) with new version(s): {matches}")
         self._web_version = matches[0]  # TODO: Update this if multiple components are available
+        self._web_date = self._getWebDate()
         self._release_notes = self._getReleaseNotes()
         self.webVersionChanged.emit()
+        self.webDateChanged.emit()
         self.releaseNotesChanged.emit()
 
         # Trigger frontend dialog opening
```
[the remark this answers is not in the document]
Comment (AndrewSazonov) [reply easyApp/Logic/Maintenance.py:165]: This was just a dummy repo to test the updater. The date is taken from the CHANGELOG.md.

**Pull request merged**